packages/esm-form-entry-app/src/app/form-submission/form-submission.service.ts

```diff
@@ -131,6 +129,47 @@ export class FormSubmissionService {
     }
   }
 
+  private confirmVisitDateAdjustment() {
+    return confirm(
+      'The encounter date falls outside the designated visit date range. Would you like to modify the visit date to accommodate the new encounter date?',
+    );
+  }
+
+  private submitEncounter(encounterCreate: EncounterCreate): Observable<any | undefined> {
+    if (!encounterCreate) {
+      return of(undefined);
+    }
+
+    if (encounterCreate.uuid) {
+      return this.encounterResourceService.getEncounterByUuid(encounterCreate.uuid).pipe(
```

Comment:
Unnecessary request to get encounter so that visit metadata is fetched?
Encounter data passed by openmrs-ngx-formentry contains no visit info. Is it worth looking into openmrs-ngx-formentry to try getting all visit data?

Reply:
In general, I think it's preferable to try to get the visit date. Since the user is required to start (or select an existing visit) before creating the form, we should already have that data without needing to fetch anything new. Maybe it can be passed in from the workspace as a property?